### PR TITLE
Docs: Update ad hoc filters docs

### PR DESCRIPTION
**Pull request by imatwawana**
Updating ad hoc filters documentation to match updated UI.
Related to https://github.com/grafana/grafana-org/issues/60 

This PR makes the following changes to the **Add variables** page:

- Updates the **Enter General options** task to match UI and link to each variable type setup task
- Updates the introduction of the **Add ad hoc filters** task and steps to match the UI
- Removes the **Create ad hoc filters** task and links to new section on the **Use dashboards** page
- Adds new **Filter dashboard data** section to **Use dashboards** page
- Adds information on editing and deleting filters 
- Updates the feature overview image on the **Use dashboards** page

**OUT OF SCOPE**: Not fixing other similar "add variable" tasks to match the updated one on this PR. Will do that on a separate PR.

### CODE REVIEW

File: docs/sources/dashboards/use-dashboards/index.md [hunk 1]
```diff
@@ -272,3 +277,30 @@ Selecting the **Auto** interval schedules a refresh based on the query time rang
...
+These filters are applied to all metric queries that use the specified data source and to all panels on the dashboard.
+
+To filter dashboard data, follow these steps:
+
+1. On the dashboard, click in the filter field.
+1. Select a label, operator, and value.
+1. Repeat this process as needed until you have all the filters you need.
+
+   ![Ad hoc filters](/media/docs/grafana/dashboards/screenshot-adhoc-filters-v11.3.png)
+
```
Comment (imatwawana) [inline docs/sources/dashboards/use-dashboards/index.md:293]: I'm missing steps on how to add multiple values to a filter, but I can't get it to work. Help!

File: docs/sources/dashboards/use-dashboards/index.md [hunk 2]
```diff
@@ -272,3 +277,30 @@ Selecting the **Auto** interval schedules a refresh based on the query time rang
 ### Control the time range using a URL
 
 {{< docs/shared lookup="dashboards/time-range-URLs.md" source="grafana" version="<GRAFANA_VERSION>" >}}
+
+## Filter dashboard data
+
+Once you've [added an ad hoc filter](ref:add-ad-hoc-filters) in the dashboard settings, you can create key/value filter pairs on the dashboard.
```
Comment (imatwawana) [inline docs/sources/dashboards/use-dashboards/index.md:283]:
```suggestion
Once you've [added an ad hoc filter](ref:add-ad-hoc-filters) in the dashboard settings, you can create label/value filter pairs on the dashboard.
```

File: docs/sources/dashboards/variables/add-template-variables/index.md [hunk 3]
```diff
@@ -233,23 +250,32 @@ groupByNode(summarize(movingAverage(apps.$app.$server.counters.requests.count, 5
 
 ## Add ad hoc filters
 
-_Ad hoc filters_ enable you to add key/value filters that are automatically added to all metric queries that use the specified data source. Unlike other variables, you do not use ad hoc filters in queries. Instead, you use ad hoc filters to write filters for existing queries.
+_Ad hoc filters_ are one of the most complex and flexible variable options available.
+Instead of a regular list of variable options, this variable allows you to build a dashboard-wide ad hoc query.
+
+Ad hoc filters let you add key/value filters that are automatically added to all metric queries that use the specified data source.
```
Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:256]:
```suggestion
Ad hoc filters let you add label/value filters that are automatically added to all metric queries that use the specified data source.
```

File: docs/sources/dashboards/use-dashboards/index.md [hunk 4]
```diff
@@ -272,3 +277,30 @@ Selecting the **Auto** interval schedules a refresh based on the query time rang
...
+
+## Filter dashboard data
+
+Once you've [added an ad hoc filter](ref:add-ad-hoc-filters) in the dashboard settings, you can create label/value filter pairs on the dashboard.
+These filters are applied to all metric queries that use the specified data source and to all panels on the dashboard.
+
+To filter dashboard data, follow these steps:
+
+1. On the dashboard, click in the filter field.
+1. Select a label, operator, and value.
```
Comment (imatwawana) [inline docs/sources/dashboards/use-dashboards/index.md:289]:
```suggestion
1. Select a label, operator, and value.

    To add multiple values for one label, choose one of the multi-select operators, **One of** (`=|`) or **Not one of** (`!=|`).

```

File: docs/sources/dashboards/use-dashboards/index.md [hunk 5]
```diff
@@ -272,3 +277,33 @@ Selecting the **Auto** interval schedules a refresh based on the query time rang
...
+
+   To add multiple values for one label, choose one of the multi-select operators, **One of** (`=|`) or **Not one of** (`!=|`).
```
Comment (imatwawana) [inline docs/sources/dashboards/use-dashboards/index.md:291]:
```suggestion
   To add multiple values for one label, choose one of the multi-select operators, **One of** (`=|`) or **Not one of** (`!=|`). These operators only appear if the filter data source supports it.
```

**Review by Sergej-Vlasov — APPROVED** [review]

**Pull request merged**